controller/appcontroller.go

```diff
@@ -489,6 +489,12 @@ func (ctrl *ApplicationController) processRequestedAppOperation(app *appv1.Appli
 		ctrl.setOperationState(app, state)
 		logCtx.Infof("Initialized new operation: %v", *app.Operation)
 	}
+
+	if state.Operation.Sync.LocalManifests != nil && app.Spec.SyncPolicy != nil {
+		logCtx.Errorf("Cannot use local sync when Automatic Sync Policy is enabled")
```

Comment:
We are catching this error too late. This should move to the API server to reject the sync request entirely. 

Also, simply logging the error and returning would not rectify the situation and the operation will be stuck in an infinite Progressing state until terminated